recipes/repo.rb

```diff
@@ -25,7 +25,9 @@
 
 # Set defaults for version, based on distribution
 node.default['hadoop']['distribution_version'] =
-  if node['hadoop']['distribution'] == 'hdp'
+  if node['hadoop'].key?('distribution_version')
+    node['hadoop']['distribution_version']
```

Comment:
I'd rather keep the assignment pretty consistent throughout the cookbook. I get what you're saying, but we do this sort of assignment everywhere else already.